### DISCUSSION TITLE
ManageMedia Permissions and SAF

### DISCUSSION THREAD

**Discussion by rusty-snake** [deckerst/aves#812] [General]
I gave aves permission to manage media. However it still requests SAF access to Pictures, DCIM, ... when I delete a file. Is media manage not enough?

**Comment by deckerst** [comment]
At the moment, Aves uses the SAF API wherever it can, and only uses the Media Store API where it can't (in practice, the `Download` folder on recent Android OS versions).

When a user gives the "manage media" permission, the OS no longer prompts the user everytime the Media Store API is used.

So, at the moment, the permission makes it only nicer in Aves when editing or moving files in the `Download` folder.

In the future, I'll reconsider whether to prefer the Media Store API when that permission is given, but the Media Store API used to be problematic in some cases and that's why the SAF API is preferred in Aves.

**Reply by rusty-snake** [reply]
> In the future, I'll reconsider whether to prefer the Media Store API when that permission is given.

Hopefully this will make aves support "system trash" then as well.

**Reply by deckerst** [reply]
Probably not, because Aves trash bin is used for media coming from the local media store but also items in vaults (which are private to the app and on different storage).
